### PR TITLE
Feature/accept datetime on retrieve

### PR DESCRIPTION
**Pull request by AlcibiadesCleinias**
resolve https://github.com/Hugovdberg/PIconnect/issues/53

But: I did not update docs. Not sure how to do it right

PS.
It is better to review and merge after https://github.com/Hugovdberg/PIconnect/pull/572 coz the pr consists of some features of 572 pr

### CODE REVIEW

File: PIconnect/time.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from datetime import datetime
+from typing import Union
+
+from PIconnect.AFSDK import AF
+
+
+def to_af_time_range(start_time: Union[str, datetime], end_time: Union[str, datetime]):
```
Comment (Hugovdberg) [inline PIconnect/time.py:7]: Could you move these type hints to comments for now? I would like to merge this for version 0.9, but that version would still need to support Python 2.7. After that release I want to add type hints as much as possible.

File: PIconnect/time.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from datetime import datetime
+from typing import Union
```
Comment (Hugovdberg) [inline PIconnect/time.py:2]: Could you remove this import for now to keep Python 2.7 support?

**Review by Hugovdberg — CHANGES_REQUESTED** [review]
I would like you to make a few changes to the new `time` module to keep Python 2.7 support for now. I like that you put this in a separate module, I think that the `PISeries.timestamp_to_index` function could be moved in there as well.

By the way, I already made a new branch `remove_python2_support` to which should be Python 3 only, and include type hints everywhere, feel free to post contributions to that branch if you want to help strip out the Python 2 compatibility. Once PIconnect 0.9 is released I want to merge that branch into `develop`.

File: PIconnect/time.py (hunk 1 above)
Comment (AlcibiadesCleinias) [reply PIconnect/time.py:7]: ok

File: PIconnect/time.py (hunk 2 above)
Comment (AlcibiadesCleinias) [reply PIconnect/time.py:2]: sure

**Comment by AlcibiadesCleinias** [comment]
>  I think that the PISeries.timestamp_to_index function could be moved in there as well.

Think the same: done

> By the way, I already made a new branch remove_python2_support to which should be Python 3 only, and include type hints everywhere, feel free to post contributions to that branch if you want to help strip out the Python 2 compatibility

Seems like a good idea. I will checkout

**Comment by AlcibiadesCleinias** [comment]
oh, ought to mention: no docs about new module generated...

**Comment by Hugovdberg** [comment]
I think not documenting the new module is fine for now, as it is meant to be used internally only. I'm working on improving the documentation structure so it becomes more clear where to add the new documentation.

**Review by Hugovdberg — APPROVED** [review]

**Pull request merged**